ci/test/assert/assert.go

```diff
@@ -0,0 +1,137 @@
+/*
+|    Protect your secrets, protect your sensitive data.
+:    Explore VMware Secrets Manager docs at https://vsecm.com/
+</
+<>/  keep your secrets... secret
+>/
+<>/' Copyright 2023-present VMware Secrets Manager contributors.
+>/'  SPDX-License-Identifier: BSD-2-Clause
+*/
+
+package assert
+
+import (
+	"fmt"
+	"github.com/pkg/errors"
+	"github.com/vmware-tanzu/secrets-manager/ci/test/io"
+	"github.com/vmware-tanzu/secrets-manager/ci/test/vsecm"
+	"github.com/vmware-tanzu/secrets-manager/ci/test/workload"
+	"strings"
+)
+
+func SentinelCanEncryptSecret(value string) error {
+	sentinel, err := vsecm.Sentinel()
+	if err != nil || sentinel == "" || value == "" {
+		return errors.Wrap(err,
+			"EncryptedSecret: Failed to define sentinel or value",
+		)
+	}
+
+	// Execute the encryption command within the sentinel pod.
+	res, err := io.Exec("kubectl", "exec", sentinel,
+		"-n", "vsecm-system", "--", "safe", "-s", value, "-e")
+	if err != nil {
+		return errors.Wrap(err, "EncryptedSecret: Failed to exec kubectl")
+	}
+
+	// Assert that the result exists.
+	if strings.TrimSpace(res) == "" {
+		return errors.New("EncryptedSecret: Encrypted secret is empty")
+	}
+
+	return nil
+}
+
+func InitContainerIsRunning() error {
+	w, err := workload.Example()
+	if err != nil || w == "" {
+		return errors.Wrap(err, "InitContainerIsRunning: Failed to define workload")
+	}
+
+	podStatus, err := io.Exec("kubectl", "get", "pod", "-n", "default", w,
+		"-o", "jsonpath={.status.initContainerStatuses[0].state.running}")
+	if err != nil {
+		return errors.Wrap(err, "InitContainerIsRunning: Failed to exec kubectl")
+	}
+
+	if podStatus == "" {
+		return errors.New("InitContainerIsRunning: Init container is not running")
+	}
+
+	return nil
+}
+
+func WorkloadIsRunning() error {
+	fmt.Println("Asserting workload is running...")
+
+	// Define the workload pod.
+	w, err := workload.Example()
+	if err != nil || w == "" {
+		return errors.Wrap(err, "WorkloadIsRunning: Failed to define workload")
+	}
+
+	// Fetch all pods in the 'default' namespace and count how many are in
+	// 'Running' status for the defined workload.
+	cmdOutput, err := io.Exec("kubectl", "get", "po", "-n", "default", "-o",
+		"jsonpath={.items[*].status.phase}")
+	if err != nil {
+		return errors.Wrap(err, "WorkloadIsRunning: Failed to exec kubectl")
+	}
+
+	// Count how many times 'Running' appears in the command output.
+	podCount := strings.Count(cmdOutput, "Running")
```

Comment:
Man! This is “way” easier to do it in bash!.
And it is more readable and expressive too.